src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -382,37 +384,30 @@ public static byte[] UrlEncodeToBytes(byte[] value, int offset, int count)
                 return null;
             }
 
-            int cSpaces = 0;
-            int cUnsafe = 0;
+            bool foundSpaces = false;
+            int unsafeCount = 0;
 
             // count them first
             for (int i = 0; i < count; i++)
             {
                 char ch = (char)value[offset + i];
 
                 if (ch == ' ')
-                    cSpaces++;
+                    foundSpaces = true;
                 else if (!IsUrlSafeChar(ch))
-                    cUnsafe++;
+                    unsafeCount++;
             }
 
             // nothing to expand?
-            if (cSpaces == 0 && cUnsafe == 0)
+            if (!foundSpaces && unsafeCount == 0)
```

Comment:
Actually, never mind; in the interests of keeping this PR short and easier to review, I'm going to submit that change in another branch. (I have one upcoming that uses `stackalloc` instead of allocating on the heap for `UrlEncode`, so I'll just merge those changes to that branch.)